lib/TokenManager.js

```diff
@@ -13,7 +14,63 @@ var cookieStorage = {
   }
 };
 
-function add(storage, key, token) {
+function clearRefreshTimeout(tokenMgmtRef, key) {
+  clearTimeout(tokenMgmtRef.refreshTimeouts[key]);
+  delete tokenMgmtRef.refreshTimeouts[key];
+}
+
+function clearRefreshTimeoutAll(tokenMgmtRef) {
+  var refreshTimeouts = tokenMgmtRef.refreshTimeouts;
+  for(var key in refreshTimeouts) {
+    if (!refreshTimeouts.hasOwnProperty(key)) {
+      continue;
+    }
+    clearRefreshTimeout(tokenMgmtRef, key);
+  }
+  tokenMgmtRef.refreshTimeouts = {};
+}
+
+function setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token) {
+  var refreshWait = (token.expiresAt * 1000) - Date.now();
+  if (refreshWait < 0) {
+    // Already expired
+    refreshWait = 0;
+  }
+  var refreshTimeout = setTimeout(function() {
+    return refresh(sdk, tokenMgmtRef, storage, key);
+  }, refreshWait);
+
+  // Clear any existing timeout
+  clearRefreshTimeout(tokenMgmtRef, key);
+
+  // Add a new timeout
+  tokenMgmtRef.refreshTimeouts[key] = refreshTimeout;
+}
+
+function setRefreshTimeoutAll(sdk, tokenMgmtRef, storage) {
+  try {
+    var tokenStorage = storage.getTokenStorage();
+  } catch(e) {
+    // Any errors thrown on instantiation will not be caught,
+    // because there are no listeners yet
+    tokenMgmtRef.emitter.emit('error', e);
+    return;
+  }
+
+  for(var key in tokenStorage) {
+    if (!tokenStorage.hasOwnProperty(key)) {
+      continue;
+    }
+    var token = tokenStorage[key];
+    if (tokenMgmtRef.autoRefresh) {
+      setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token);
+    } else if (token.expiresAt * 1000 < Date.now()) {
```

Comment:
We might want to deal with this in another PR, but what happens in the case of no autoRefresh and the token hasn't expired on init? It seems like when that token expires, it won't emit an 'expired' message.